staticaddr/staticutil/utils.go

```diff
@@ -0,0 +1,205 @@
+	"fmt"
+	"sort"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightninglabs/loop/staticaddr/address"
+	"github.com/lightninglabs/loop/staticaddr/deposit"
+	"github.com/lightninglabs/loop/staticaddr/script"
+	"github.com/lightninglabs/loop/swapserverrpc"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnwallet"
+)
+
+// ToPrevOuts converts a slice of deposits to a map of outpoints to TxOuts.
+func ToPrevOuts(deposits []*deposit.Deposit,
+	pkScript []byte) (map[wire.OutPoint]*wire.TxOut, error) {
+
+	prevOuts := make(map[wire.OutPoint]*wire.TxOut, len(deposits))
+	for _, d := range deposits {
+		outpoint := wire.OutPoint{
+			Hash:  d.Hash,
+			Index: d.Index,
+		}
+		txOut := &wire.TxOut{
+			Value:    int64(d.Value),
+			PkScript: pkScript,
+		}
+		if _, ok := prevOuts[outpoint]; ok {
+			return nil, fmt.Errorf("duplicate outpoint %v",
+				outpoint)
+		}
+		prevOuts[outpoint] = txOut
+	}
+
+	return prevOuts, nil
+}
+
+// CreateMusig2Sessions creates a musig2 session for a number of deposits.
+func CreateMusig2Sessions(ctx context.Context,
+	signer lndclient.SignerClient, deposits []*deposit.Deposit,
+	addrParams *address.Parameters,
+	staticAddress *script.StaticAddress) ([]*input.MuSig2SessionInfo,
+	[][]byte, error) {
+
+	musig2Sessions := make([]*input.MuSig2SessionInfo, len(deposits))
+	clientNonces := make([][]byte, len(deposits))
+
+	// Create the sessions and nonces from the deposits.
+	for i := 0; i < len(deposits); i++ {
+		session, err := CreateMusig2Session(
+			ctx, signer, addrParams, staticAddress,
+		)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		musig2Sessions[i] = session
+		clientNonces[i] = session.PublicNonce[:]
+	}
+
+	return musig2Sessions, clientNonces, nil
+}
+
+// CreateMusig2SessionsPerDeposit creates a musig2 session for a number of
+// deposits.
+func CreateMusig2SessionsPerDeposit(ctx context.Context,
+	signer lndclient.SignerClient, deposits []*deposit.Deposit,
+	addrParams *address.Parameters,
+	staticAddress *script.StaticAddress) (
+	map[string]*input.MuSig2SessionInfo, map[string][]byte, map[string]int,
+	error) {
+
+	sessions := make(map[string]*input.MuSig2SessionInfo)
+	nonces := make(map[string][]byte)
+	depositToIdx := make(map[string]int)
+
+	// Create the musig2 sessions for the sweepless sweep tx.
+	for i, deposit := range deposits {
+		session, err := CreateMusig2Session(
+			ctx, signer, addrParams, staticAddress,
+		)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		sessions[deposit.String()] = session
+		nonces[deposit.String()] = session.PublicNonce[:]
+		depositToIdx[deposit.String()] = i
+	}
+
+	return sessions, nonces, depositToIdx, nil
+}
+
+// CreateMusig2Session creates a musig2 session for the deposit.
+func CreateMusig2Session(ctx context.Context,
+	signer lndclient.SignerClient, addrParams *address.Parameters,
```

Comment:
Idea: can we move address.Parameters to `script` package and dedup with its counterpart on naut side? Then we can dedup this function with naut as well.

Reply:
I will refactor this in a follow-up PR. Created issue https://github.com/lightninglabs/loop/issues/1056